runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/util/JMESPath.kt

```diff
@@ -62,8 +62,7 @@ public fun Any?.type(): String = when (this) {
     is List<*>, is Array<*> -> "array"
     is Number -> "number"
     is Any -> "object"
-    null -> "null"
-    else -> throw Exception("Undetected type for: $this")
+    else -> "null"
```

Comment:
question: was this causing issues anywhere?

Reply:
yeah, it was a warning saying the "else" block was unreachable

Reply:
I was having trouble replicating the exact warning locally just now, so I reverted the change